subprojects/integ-tests/src/test/kotlin/org/gradle/kotlin/dsl/integration/ProjectSchemaAccessorsIntegrationTest.kt

```diff
@@ -129,41 +117,83 @@ class ProjectSchemaAccessorsIntegrationTest : AbstractPluginIntegrationTest() {
 
             println("extension: " + typeOf(extension))
             extension { println("extension{} " + typeOf(this)) }
-
-            println("beans: " + typeOf(beans))
-            beans { println("beans{} " + typeOf(beans)) }
         """)
 
         assertThat(
             build("help", "-q").output,
             containsMultiLineString("""
                 extension: Extension
                 extension{} Extension
-                beans: org.gradle.api.NamedDomainObjectContainer<Extension>
-                beans{} org.gradle.api.NamedDomainObjectContainer<Extension>
             """)
         )
     }
 
     @Test
-    fun `can access task of default package type`() {
+    fun `can access container element of default package type`() {
 
-        withFolders {
+        withBuildSrc {
 
-            "buildSrc" {
+            "src/main/kotlin" {
 
-                withPrecompiledPlugins()
+                withFile("Bean.kt", """
+                    open class Bean(private val name: String) : org.gradle.api.Named {
+                        override fun getName() = name
+                    }
+                """)
 
-                "src/main/kotlin" {
+                withFile("plugin.gradle.kts", """
 
-                    withFile("CustomTask.kt", """
-                        open class CustomTask : org.gradle.api.DefaultTask()
-                    """)
+                    import org.gradle.api.internal.*
+                    import org.gradle.internal.reflect.*
 
-                    withFile("plugin.gradle.kts", """
-                        tasks.register<CustomTask>("customTask")
-                    """)
-                }
+                    val beans = DefaultPolymorphicDomainObjectContainer(
+                        Any::class.java,
+                        DirectInstantiator.INSTANCE
+                    ) { (it as Bean).name }
+
+                    // TODO: schema doesn't expose the registered element as Bean
```

Comment:
This looks like a bug to me, polymorphic containers are supposed to expose the implementation type in their schema.